### DISCUSSION TITLE
How to generate flamegraph for API request?

### DISCUSSION THREAD

**Discussion by illeatmyhat** [fastapi/fastapi#10766] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def root():
    return {"message": "Hello World"}
```

### Description

I have an application that runs dramatically slower when deployed, and I'd like to be able to generate flamegraphs to see where the performance bottlenecks are, and if they're different than what I get on my local machine.

I can get flamegraph reports for isolated functions, but when running the whole app using hypercorn, I get the error
```
[2023-12-07T18:42:52.267900000Z ERROR inferno::flamegraph] No stack counts found
Error: Failed to write flamegraph: I/O error: No stack counts found
```
How can I get flamegraph reports for individual API calls?

### Operating System

macOS

### Operating System Details

OSX 14.0
MacBook Pro 2021 M1 Max

### FastAPI Version

0.99.1

### Pydantic Version

1.10.12

### Python Version

Python 3.9.16

### Additional Context

I've attempted using py-spy, pyinferno, and cargo-flamegraph

**Comment by jgould22** [comment]
I do not have an answer to your question other than to say when you start using a process that spawns multiple workers which are event loops flame graphs can get a bit finicky to generate.

Generally when trying to solve the problem you describe people will reach for tooling for tracing like open-telemtry. Unfortunately this is not nearly as easy as generating a flame graph.
